File: leetcode/1768-merge_strings_alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        final_array = []
        len1, len2 = len(word1), len(word2)
        max_len = max(len1, len2)
        
        for i in range(max_len):
            if i < len1:
                final_array.append(word1[i])
            if i < len2:
                final_array.append(word2[i])
        
        return ''.join(final_array)

File: solutions/leetcode/1768-merge_strings_alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int m = word1.length();
        int n = word2.length();

        StringBuilder result = new StringBuilder();
        int i = 0;
        int j = 0;

        while(i < m || j < n){
            if (i < m){
                result.append(word1.charAt(i));
                i++;
            } 
            if (j < n){
                result.append(word2.charAt(j));
                j++;
            }
        }
        return result.toString();
    }
}